src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -532,8 +554,20 @@ public void validateContentType(FailureCollector failureCollector) {
       }
     }
 
-    public String getBucket() {
-      return GCSPath.from(path).getBucket();
+    /**
+     * Get the bucket name from the path.
+     * @param collector failure collector
+     * @return bucket name as {@link String} if found, otherwise null.
+     */
+    public String getBucket(FailureCollector collector) {
+      try {
+        return GCSPath.from(path).getBucket();
+      } catch (IllegalArgumentException e) {
```

Comment:
Do we only want to log for `IllegalArgumentException` ? or it's the only expected one ?

Reply:
Synced offline. It throws only IllegalArgumentException